### ISSUE TITLE
Nested scripts issue new to 4.1.0-beta2

### ISSUE THREAD

**Issue by joeledge** [chrismaltby/gb-studio#1534]
**Describe the bug**
Prior to version 4.1.0-beta2, a script I used to check if the player had triggered a custom encounter before or not worked, but stopped working from this version.

The script included two scripts within it to check flags (as I use flags often, I'd made custom scripts to repeat the code less), one script to add a flag to a variable based on a value passed to the custom script, one script to check a variable has a flag again based on a value passed to the custom script.

I've attached a blank project with this happening.

**To Reproduce**
1. Go to Scene 1 and build
2. Run to your right to the end of the screen, on leaving the trigger it will let you know a flag has been added with **'Added Flag 4'.**
3. Leaving the trigger a second time should let you know the flag has already been added with 'Flag 4 already set' if it's working, or it'll say the flag has been added again **'Added Flag 4'**. if it hasn't worked.
4. To get it to work, disable the custom script 'Have I triggered this before' and enable the disabled 'Less nested event works, below doesn't' script group. This will work as intended, but break once nested another level.

**Expected behavior**
Previously, the script with two nested events worked and set the flag.

**Screenshots**
![Screenshot 2024-08-24 at 15 37 02](https://github.com/user-attachments/assets/1440c91b-122d-4552-8438-bb810a5fc394)
![Screenshot 2024-08-24 at 15 36 51](https://github.com/user-attachments/assets/9010a39c-3724-4466-b6bc-b3f501d199de)

**Platform (please complete the following information):**
 - OS: MacOS 14.5
 - GB Studio version 4.1.0-beta2

**Additional context**
Attached sample project.

[Debug Nested Includes.zip](https://github.com/user-attachments/files/16736953/Debug.Nested.Includes.zip)

**Comment by chrismaltby** [comment]
Hi @joeledge Thanks for this, using this example project I believe I've managed to get a fix in place.

Your example project actually did have a small bug inside the "Have I triggered this before?" script
<img width="386" alt="Screenshot 2024-08-27 at 17 59 54" src="https://github.com/user-attachments/assets/a630705b-af23-44d0-980e-38c85bf13a00">
should have been this:
<img width="392" alt="Screenshot 2024-08-27 at 17 58 07" src="https://github.com/user-attachments/assets/e283b80b-c93b-48a4-86fb-ee4117ed2ae2">

but even with this fixed there was still the problem.

The fixed build will be available here: https://github.com/chrismaltby/gb-studio/actions/runs/10582432095

I can confirm that trying your project with the previous build I saw encounters triggering multiple times but with this latest build they only trigger once.
